x-pack/plugins/fleet/server/mocks/index.ts

```diff
@@ -113,33 +132,42 @@ export const createMockAgentPolicyService = (): jest.Mocked<AgentPolicyServiceIn
 /**
  * Creates a mock AgentService
  */
-export const createMockAgentService = (): jest.Mocked<AgentService> => {
+export const createMockAgentService = () => agentServiceMock.create();
+
+/**
+ * Creates a mock AgentClient
+ */
+export const createMockAgentClient = () => agentServiceMock.createClient();
+
+export const createMockPackageService = (): PackageService => {
   return {
-    getAgentStatusById: jest.fn(),
-    getAgentStatusForAgentPolicy: jest.fn(),
-    getAgent: jest.fn(),
-    listAgents: jest.fn(),
+    getInstallation: jest.fn(),
+    ensureInstalledPackage: jest.fn(),
   };
 };
 
 /**
  * Creates mock `authz` object
  */
-export const fleetAuthzMock: FleetAuthz = {
-  fleet: {
-    all: true,
-    setup: true,
-    readEnrollmentTokens: true,
-  },
-  integrations: {
-    readPackageInfo: true,
-    readInstalledPackages: true,
-    installPackages: true,
-    upgradePackages: true,
-    removePackages: true,
-    readPackageSettings: true,
-    writePackageSettings: true,
-    readIntegrationPolicies: true,
-    writeIntegrationPolicies: true,
-  },
+export const createFleetAuthzMock = (): FleetAuthz => {
+  return {
+    fleet: {
+      all: true,
+      setup: true,
+      readEnrollmentTokens: true,
+    },
+    integrations: {
+      readPackageInfo: true,
+      readInstalledPackages: true,
+      installPackages: true,
+      upgradePackages: true,
+      removePackages: true,
+      readPackageSettings: true,
+      writePackageSettings: true,
+      readIntegrationPolicies: true,
+      writeIntegrationPolicies: true,
+    },
+  };
 };
+
+export const fleetAuthzMock: FleetAuthz = createFleetAuthzMock();
```

Comment:
FYI - I meant to remove this `const` from here, when I created the above function, since it might create side affects being that it is mutable, but never got around to it.